boards: Update STM32 disco boards default configuration

erwango opened the pull request:
First batch of patches to address #9247.

Impacted boards:
-stm32l496g_disco
-disco_l475_iot1
-stm32l476g_disco
-stm32f769i_disco
-stm32f723e_disco
-stm32f469i_disco
-stm32f429i_disco

erwango committed:
boards: disco_l475_iot1: Conform to default configuration guidelines

Update disco_l475_iot1 configuration to match with
default configuration guidelines:
-Deactivate features by default
-Configure available connectors

Signed-off-by: Erwan Gouriou <erwan.gouriou@linaro.org>

erwango committed:
boards: stm32l496g_disco: Conform to default configuration guidelines

Update stm32l496g_disco configuration to match with
default configuration guidelines:
-Deactivate features by default
-Configure available connectors

Signed-off-by: Erwan Gouriou <erwan.gouriou@linaro.org>

erwango committed:
boards: stm32f769i_disco: Conform to default configuration guidelines

Update stm32f769i_disco configuration to match with
default configuration guidelines:
-Configure available connectors

Signed-off-by: Erwan Gouriou <erwan.gouriou@linaro.org>

erwango committed:
boards: stm32f723e_disco: Conform to default configuration guidelines

Update stm32f723e_disco configuration to match with
default configuration guidelines:
-Configure available connectors

Signed-off-by: Erwan Gouriou <erwan.gouriou@linaro.org>

erwango committed:
boards: stm32l476g_disco: update yaml file

Update stm32l476g_disco yaml file

Signed-off-by: Erwan Gouriou <erwan.gouriou@linaro.org>

ydamigos reviewed (changes requested):
Looks good. Some minor comments.

erwango committed:
boards: stm32f469i_disco: Conform to default configuration guidelines

Update stm32f469i_disco configuration to match with
default configuration guidelines:
-Configure available connectors
-Update yaml file

Signed-off-by: Erwan Gouriou <erwan.gouriou@linaro.org>

erwango commented on boards/arm/stm32f469i_disco/doc/stm32f469i_disco.rst:
```diff
@@ -115,10 +115,12 @@ For mode details please refer to `32F469IDISCOVERY board User Manual`_.
+- UART_3 TX/RX : PB10/PB11 (ST-Link Virtual Port Com)
+- UART_6 TX/RX : PG14/PG9 (Arduino Serial)
+- I2C1 SCL/SDA : PB8/PB9 (Arduino I2C)
+- SPI1 NSS/SCK/MISO/MOSI : PA4/PA5/PA6/PA7 (Arduino SPI)
```
Fixed

erwango commented on boards/arm/stm32f769i_disco/doc/stm32f769i_disco.rst:
```diff
@@ -126,10 +130,11 @@ For mode details please refer to `32F769I-DISCO board User Manual`_.
+- UART_1 TX/RX : PA9/PA10 (ST-Link Virtual Port Com)
+- UART_6 TX/RX : PC6/PC7 (Arduino Serial)
+- I2C1 SCL/SDA : PB8/PB9 (Arduino I2C)
+- SPI1 SCK/MISO/MOSI : PA12/PB14/PB15 (Arduino SPI)
```
Fixed

erwango commented:
@ydamigos , comments fixed. Thanks for the review

dbkinder reviewed (approved):
doc changes look OK.
